src/sftp/handler.go

```diff
@@ -258,11 +262,12 @@ func (receiver *SftpHandler) copySingleFile(fileInfo os.FileInfo, index int, dir
 		}
 	}
 
-	err = receiver.sftpClient.Remove(fullFilePath)
-	if err != nil {
-		slog.Error("Failed to remove file from SFTP server", slog.Any(utils.ErrorKey, err), slog.String(utils.FileNameKey, fullFilePath))
-		return
+	if !isZip || deleteZip {
```

Comment:
It would be nice to have a test around this new condition

Reply:
I ran code with coverage and it say's that it is. I believe it's covered but just making sure before I merge/add more tests.
<img width="1035" alt="Screenshot 2024-11-07 at 12 01 55 PM" src="https://github.com/user-attachments/assets/21717756-e09d-4b94-90da-a3a921032d40">

Reply:
I do see other scenarios that aren't covered that aren't part of the `if !isZip || deleteZip`. If that's what you meant, then I can look into adding coverage for those.

Reply:
Added an assert for making sure **Remove** is not called when a zip file is not successfully unzipped.